packages/monitor-v2/src/bot-oo-v3/SettleAssertions.ts

```diff
@@ -0,0 +1,43 @@
+import { getContractInstanceWithProvider, Logger, MonitoringParams, OptimisticOracleV3Ethers } from "./common";
+import { logSettleAssertion } from "./BotLogger";
+
+export async function settleAssertions(logger: typeof Logger, params: MonitoringParams): Promise<void> {
+  const oo = await getContractInstanceWithProvider<OptimisticOracleV3Ethers>("OptimisticOracleV3", params.provider);
+
+  const assertions = await oo.queryFilter(oo.filters.AssertionMade(), undefined, undefined);
+  const assertionsSettled = await oo.queryFilter(oo.filters.AssertionSettled(), undefined, undefined);
```

Comment:
Is this script expected to run on any chain? I don't think direct queries from block 0 are allowed by any provider X chain except for Infura on mainnet. Would suggest that we break these queries up.

[Here's](https://github.com/across-protocol/relayer-v2/blob/0e1a06381f0d56ca084228b3382b51ed0f203343/src/utils/EventUtils.ts#L35-L120) some code where this is done in across -- there are some parts that are probably not needed for this use case, but just wanted to link it in case it came in handy.